climada_petals/hazard/wildfire.py

```diff
@@ -1105,7 +1144,15 @@ def _run_one_fire(self):
                 centr_burned = self._fire_propagation_on_matrix(self.centroids.shape, \
                     self.centroids.fire_propa_matrix, self.ProbaParams.prop_proba, \
                     centr_ix, centr_iy, centr_burned, np.random.random(500))
-
+            
+            else: # added by Sam G.
+                centr_ix, centr_iy = burned[np.random.randint(0, len(burned))] # added by Sam G.
+                num_tries += 1 # added by Sam G.
+                if num_tries == 1000: # added by Sam G.
+                    LOGGER.info('Propagation conditions are not met') # added by Sam G.
```

Comment:
Is this really an info? Should this not raise an error? What happens if the number of trials is above 1000? Also, is it really the best algorithmic way to solve this?